src/omotes_orchestrator/timeout_job_manager.py

```diff
@@ -0,0 +1,126 @@
+                 postgresql_if: PostgresInterface,
+                 orchestrator: Optional[Orchestrator],
+                 config: TimeoutJobManagerConfig) -> None:
+        """Construct the timeout job manager."""
+        self.postgresql_if = postgresql_if
+        self.orchestrator = orchestrator
+
+        self._init_time = datetime.now()
+        self._start_buffer_sec = config.start_buffer_sec
+        self._rerun_sec = config.rerun_sec
+
+        self._stop_event = threading.Event()
+        self._timeout_jobs_handler_thread = threading.Thread(target=self.timeout_jobs_handler)
+        self._timeout_jobs_handler_thread_active = False
+
+    def start(self) -> None:
+        """Start the timeout job manager as a daemon process."""
+        if not self._timeout_jobs_handler_thread_active:
+            LOGGER.info("Starting the timeout job manager")
+
+            self._stop_event.clear()
+            self._timeout_jobs_handler_thread.start()
+            self._timeout_jobs_handler_thread_active = True
+
+    def stop(self) -> None:
+        """Stop the timeout job manager."""
+        if self._timeout_jobs_handler_thread_active:
+            LOGGER.info("Stopping the timeout job manager")
+
+            self._stop_event.set()
+            self._timeout_jobs_handler_thread.join(timeout=5.0)
+            self._timeout_jobs_handler_thread_active = False
+
+    def _system_activation_sec(self) -> float:
+        """Return period in seconds since the TimeoutJobManager instantiated."""
+        return (datetime.now() - self._init_time).total_seconds()
+
+    def timeout_jobs_handler(self) -> None:
+        """Start a background process to cancel timed out jobs.
+
+        The function periodically checks if there are any timed out jobs.
+        Meanwhile, if the TimeoutJobManager is instantiated longer than the
+        configured buffer time, and the relationship between TimeoutJobManager
+        and Orchestrator is correctly linked (self.orchestrator is not None, so
+        self.orchestrator.job_cancellation_handler() can be called),
+        the job/row can be canceled outright.
+        """
+        while not self._stop_event.is_set():
+            if self.orchestrator is None:
+                LOGGER.error("Exiting timeout_jobs_handler(): orchestrator is None.")
+                break
+
+            if self._system_activation_sec() > self._start_buffer_sec:
+                jobs = self.postgresql_if.get_all_jobs()
+                for job in jobs:
+                    if self.job_is_timedout(job):
+                        self.orchestrator.job_cancellation_handler(JobCancel(uuid=str(job.job_id)))
+
+                        timeout_job_manager_up_mins = round(self._system_activation_sec() / 60, 1)
+                        LOGGER.warning("TimeoutJobManager is up %s mins. "
+                                       + "Found and canceled a timed out job %s",
+                                       timeout_job_manager_up_mins, job.job_id)
+
```

Comment:
Perhaps we could add a single log line here where it logs that it processed `len(jobs)` to check for timed out jobs in the INFO level. Just to ensure that the system notifies that it is doing something.

Reply:
Hhmm an alternative could also be to lower the log level to DEBUG. What would be your preference?

Reply:
Updated to debug level